src/Bicep.Core/Syntax/ResourceAccessSyntax.cs

```diff
@@ -0,0 +1,31 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+using Bicep.Core.Navigation;
+using Bicep.Core.Parsing;
+
+namespace Bicep.Core.Syntax
+{
+    public class ResourceAccessSyntax : ExpressionSyntax, ISymbolReference
+    {
+        public ResourceAccessSyntax(SyntaxBase baseExpression, Token colon, IdentifierSyntax resourceName)
+        {
+            AssertTokenType(colon, nameof(colon), TokenType.Colon);
+
+            this.BaseExpression = baseExpression;
+            this.Colon = colon;
+            this.ResourceName = resourceName;
+        }
+
+        public SyntaxBase BaseExpression { get; }
+
+        public Token Colon { get; }
+
+        public IdentifierSyntax ResourceName { get; }
+
+        IdentifierSyntax ISymbolReference.Name => ResourceName;
```

Comment:
Is this going to work for consumers of `ISymbolReference`? Will existing users correctly understand that this isn't a globally-scoped symbol?

Reply:
That's a good question that I don't have the answer to ... I found some issues in the integration tests for tooling that I haven't investigated yet.

Reply:
Ok, so now that I've fixed all of the integration tests. I haven't found cases where anyone assumes that this is for global symbols. I did find cases (tests) where they were special case things instead of using `ISymbolReference`.

I tested features like find all references, rename, gtd, etc, and all working well.